pkg/appclient/relay_verifier.go

```diff
@@ -0,0 +1,47 @@
+package appclient
+
+import (
+	"context"
+
+	"github.com/cometbft/cometbft/crypto"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"pocket/x/service/types"
+)
+
+// verifyResponse verifies the relay response signature.
+func (app *appClient) verifyResponse(
+	ctx context.Context,
+	supplierAddress string,
+	relayResponse *types.RelayResponse,
+) error {
+	// Query for the supplier account to get the application's public key to verify the relay request signature.
+	accQueryReq := &accounttypes.QueryAccountRequest{Address: supplierAddress}
+	accQueryRes, err := app.accountQuerier.Account(ctx, accQueryReq)
+	if err != nil {
+		return err
+	}
+
+	// Marshal the relay response payload to bytes and get the hash.
+	var payloadBz []byte
+	_, err = relayResponse.Payload.MarshalTo(payloadBz)
+	if err != nil {
+		return err
+	}
+	hash := crypto.Sha256(payloadBz)
```

Comment:
I think there should be a small help on `RelayResponse` that removes the necessary fields and takes in helpers supplied by the appclient to do the actual signature verification.

Other than that, the compartmentalization is great